src/spartan/polys/multilinear.rs

```diff
@@ -283,8 +284,65 @@ mod tests {
 
   #[test]
   fn test_evaluation() {
-    test_evaluation_with::<Fp>();
+    test_evaluation_with::<pasta_curves::Fp>();
     test_evaluation_with::<provider::bn256_grumpkin::bn256::Scalar>();
     test_evaluation_with::<provider::secp_secq::secp256k1::Scalar>();
   }
+
+  /// Returns a random ML polynomial
+  fn random<R: RngCore + CryptoRng, Scalar: PrimeField>(
+    num_vars: usize,
+    mut rng: &mut R,
+  ) -> MultilinearPolynomial<Scalar> {
+    MultilinearPolynomial::new(
+      std::iter::from_fn(|| Some(Scalar::random(&mut rng)))
+        .take(1 << num_vars)
+        .collect(),
+    )
+  }
+
+  /// This binds the variables of a multilinear polynomial to a provided sequence
+  /// of values.
+  /// 
+  /// Assuming `bind_poly_var_top` defines the "top" variable of the polynomial,
+  /// this aims to test whether variables should be provided to the 
+  /// `evaluate` function in topmost-first (big endian) of topmost-last (lower endian)
+  /// order.
+  fn bind_sequence<F: PrimeField>(
+    poly: &MultilinearPolynomial<F>,
+    values: &[F],
+  ) -> MultilinearPolynomial<F> {
+    // Assert that the size of the polynomial being evaluated is a power of 2 greater than (1 << values.len())
+    assert!(poly.Z.len().is_power_of_two());
+    assert!(poly.Z.len() >= 1 << values.len());
+
+    let mut tmp = poly.clone();
+    for v in values.iter() {
+      tmp.bind_poly_var_top(v);
+    }
+    tmp
+  }
+
+  fn partial_and_evaluate_with<F: PrimeField>() {
```

Comment:
nit: `partial` --> `bind` for improved consistency?

Reply:
Renamed to the two functions which output is compared in the test.

Reply:
thanks!